# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/Marking.cpp
# Commit: 4746d15d7a9f
# Full Hash: 4746d15d7a9fa501c702bd1437d3f3da127a86ee
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-10-28 04:43:55
# Regressor Bug: 1797755
# File Overlap Count: 1
# Description:
#   Bug 1797755 - Part 3: Remove the unused MarkStackIter class r=sfink
#   
#   Depends on D160525
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D160526
# ==============================================================================

diff -r 8be86d9e2f18 -r 4746d15d7a9f js/src/gc/Marking.cpp
--- a/js/src/gc/Marking.cpp	Thu Oct 27 18:23:02 2022 +0000
+++ b/js/src/gc/Marking.cpp	Thu Oct 27 18:23:03 2022 +0000
@@ -1716,51 +1716,6 @@
   return stack().sizeOfExcludingThis(mallocSizeOf);
 }
 
-MarkStackIter::MarkStackIter(MarkStack& stack)
-    : stack_(stack), pos_(stack.position()) {
-#ifdef DEBUG
-  stack.iteratorCount_++;
-#endif
-}
-
-MarkStackIter::~MarkStackIter() {
-#ifdef DEBUG
-  MOZ_ASSERT(stack_.iteratorCount_);
-  stack_.iteratorCount_--;
-#endif
-}
-
-inline size_t MarkStackIter::position() const { return pos_; }
-
-inline bool MarkStackIter::done() const { return position() == 0; }
-
-inline MarkStack::TaggedPtr MarkStackIter::peekPtr() const {
-  MOZ_ASSERT(!done());
-  return stack_.stack()[pos_ - 1];
-}
-
-inline MarkStack::Tag MarkStackIter::peekTag() const { return peekPtr().tag(); }
-
-inline void MarkStackIter::nextPtr() {
-  MOZ_ASSERT(!done());
-  MOZ_ASSERT(!TagIsRangeTag(peekTag()));
-  pos_--;
-}
-
-inline void MarkStackIter::next() {
-  if (TagIsRangeTag(peekTag())) {
-    nextArray();
-  } else {
-    nextPtr();
-  }
-}
-
-inline void MarkStackIter::nextArray() {
-  MOZ_ASSERT(TagIsRangeTag(peekTag()));
-  MOZ_ASSERT(position() >= ValueRangeWords);
-  pos_ -= ValueRangeWords;
-}
-
 /*** GCMarker ***************************************************************/
 
 /*
